apiserver/src/main/java/org/dependencytrack/resources/v1/PermissionResource.java

```diff
@@ -264,4 +269,61 @@ public Response removePermissionFromTeam(
             return Response.status(Response.Status.NOT_MODIFIED).build();
         }
     }
+
+    @PUT
+    @Path("/user/{username}")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(description = "<p>Requires permission <strong>ACCESS_MANAGEMENT</strong> or <strong>ACCESS_MANAGEMENT_DELETE</strong></p>")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "The updated user", content = @Content(schema = @Schema(implementation = Team.class))),
+            @ApiResponse(responseCode = "401", description = "Unauthorized"),
+            @ApiResponse(responseCode = "404", description = "The user could not be found")
+    })
+    @PermissionRequired({ Permissions.Constants.ACCESS_MANAGEMENT, Permissions.Constants.ACCESS_MANAGEMENT_DELETE })
+    public Response setUserPermissions(
+            @Parameter(description = "A valid user uuid", schema = @Schema(type = "string", format = "uuid"), required = true) @PathParam("username") String username,
+            @Parameter(description = "A valid permission", required = true) PermissionsSetRequest request) {
+        try (QueryManager qm = new QueryManager()) {
+            UserPrincipal user = qm.getUserPrincipal(username);
+            if (user == null) {
+                return Response.status(Response.Status.NOT_FOUND).entity("The user could not be found.").build();
+            }
+
+            final List<Permission> requestedPermissions = request.permissions()
+                    .stream()
+                    .map(qm::getPermission)
+                    .toList();
+
+            // check that all requested teams exist
+            List<String> notFound = new ArrayList<>();
+            for (int i = 0; i < requestedPermissions.size(); i++) {
+                if (requestedPermissions.get(i) == null) {
+                    notFound.add(request.permissions().stream().toList().get(i));
+                }
```

Comment:
You might get unexpected results from this. `request.permissions` is a `Set`, which does not guarantee ordering. The chance is quite high that upon each invocation of `request.permissions().stream().toList()`, you'll get a list in different order.